internal/docstore/dynamodocstore/codec.go

```diff
@@ -140,8 +140,8 @@ func encodeFloat(f float64) *dyn.AttributeValue {
 
 ////////////////////////////////////////////////////////////////
 
-func decodeDoc(doc driver.Document, item map[string]*dyn.AttributeValue) error {
-	return doc.Decode(decoder{&dyn.AttributeValue{M: item}})
+func decodeDoc(doc driver.Document, item *dyn.AttributeValue) error {
```

Comment:
nit: reverse argument order, because by convention Decode/Unmarshal functions take the destination as a second arg.

Reply:
Done.